spec/ddtrace/tracer_integration_spec.rb

```diff
@@ -76,4 +76,53 @@ def lang_tag(span)
       it { expect(@child_root_span).to be child_span }
     end
   end
+
+  context 'with synthetics' do
+    context 'which applies the context from distributed tracing headers' do
+      let(:trace_id) { 3238677264721744442 }
+      let(:parent_id) { 0 }
+      let(:sampling_priority) { 1 }
+      let(:origin) { 'synthetics' }
+
+      let(:distributed_tracing_headers) do
+        {
+          rack_header(Datadog::Ext::DistributedTracing::HTTP_HEADER_TRACE_ID) => trace_id.to_s,
```

Comment:
I don't like that the integration test for the tracer knows about Rack-formatted headers; again, represents coupling that should not be present. The necessity of this code is a strong suggestion that we should refactor the common propagation mechanism to be agnostic to header format: the integration in question should be responsible for doing any mapping, if necessary.

Reply:
Added a TODO to the appropriate application code.